crates/energy/src/battery.rs

```diff
@@ -1,10 +1,15 @@
 use bevy::prelude::*;
+use de_core::objects::Active;
+use de_spawner::SpawnerSet;
 
 pub(crate) struct BatteryPlugin;
 
 impl Plugin for BatteryPlugin {
     fn build(&self, app: &mut App) {
-        app.add_systems(Update, discharge_battery);
+        app.add_systems(
+            Update,
+            (discharge_battery, spawn_battery.after(SpawnerSet::Spawn)),
```

Comment:
Ditto, `spawn_battery()` should be executed during `PostUpdate` as the spawned entity is not available immediately during `Update`. Therefore, `SpawnerSet::Spawn` is not even needed.